File: 2024/go/day1.go
package main

import (
    "fmt"
	"os"
	"strings"
	"slices"
	"math"
)

func main() {
    var data, err = os.ReadFile("assets/1-1_input.txt")
    if err != nil {
        fmt.Println("Cannot open file");
    }

    var input string = string(data)
    input = strings.Trim(input, "\n")

    var input_lines []string = strings.Split(input, "\n")
    var (left []string; right []string)

    for _, val := range input_lines {
        var split = strings.Split(val, " ")
        left = append(left, split[0])
        right = append(right, split[len(split)-1])
    }

    result := 0

    for _, val := range slices.Sorted(left) {
        j := -6

        if j != -1 {
            result += int(math.Abs(float64(i - j)))
            right[j] = "noon"
        }
    }

    fmt.Println("Result:", result)
}

File: 2024/rust/src/bin/day1.rs
use std::{fs::File, io::Read};

fn main() {
    let buf = {
        let mut buf = String::new();
        File::open("assets/1-1_input.txt")
            .unwrap()
            .read_to_string(&mut buf)
            .unwrap();
        buf
    };

    let pairs: Vec<(_, _)> = buf
        .trim()
        .split('\n')
        .filter_map(|line| line.split_once("   "))
        .collect();

    let (left, right): (Vec<_>, Vec<_>) = pairs.clone().into_iter().unzip();

    for (val1, val2) in pairs {
        println!("{val1}::{val2}")
    }

    println!("{}", left.len());
    println!("{}", right.len());
}
